lib/translator/inquiry.cc

```diff
@@ -17,6 +17,48 @@
 
 namespace translator {
 
+  // anonymous namepace for helper functions that are used in this file
+  namespace {
+
+    // The SPT field of the ExtendedInquiryData depends on the DPC field of the
+    // IdentifyNamespaceData
+    uint8_t GetSptValueFromDPC(
+      const nvme_defs::IdentifyNamespace &identify_namespace_data) {
+      
+      uint8_t dpc = 0;
+      dpc |= identify_namespace_data.dpc.pit3;
+      dpc |= identify_namespace_data.dpc.pit2 << 1;
+      dpc |= identify_namespace_data.dpc.pit1 << 2;
+      
+      uint8_t spt;
+  
+      switch (dpc) {
```

Comment:
Please document this switch statement; it's completely opaque what's going on here.

Reply:
Added the link to the NVME translation spec